cmd/deploy/deploy_test.go

```diff
@@ -196,4 +209,73 @@ func TestDeploy(t *testing.T) {
+		appDir := t.TempDir()
+		test.CopyDir(t, "../../testdata/streamlit_app", appDir)
+		expectedVersion := "v1.2.3"
+		expectedMessage := "expected message"
+		apps := mockAppExistsWithDeployEventsRun(func(args mock.Arguments) {
+			input := args.Get(1).(app.DeployEventsInput)
+			input.Handler(app.DeployEvent{Typename: "AppBuildMessageEvent", BuildMessage: app.AppBuildMessageEvent{Message: "Build message 1"}})    // nolint:errcheck
+			input.Handler(app.DeployEvent{Typename: "AppBuildMessageEvent", BuildMessage: app.AppBuildMessageEvent{Message: "Build message 2"}})    // nolint:errcheck
+			input.Handler(app.DeployEvent{Typename: "AppDeploymentStatusEvent", DeploymentStatus: app.AppDeploymentStatusEvent{Status: "PENDING"}}) // nolint:errcheck
+			input.Handler(app.DeployEvent{Typename: "AppDeploymentStatusEvent", DeploymentStatus: app.AppDeploymentStatusEvent{Status: "PENDING"}}) // nolint:errcheck
+			input.Handler(app.DeployEvent{Typename: "AppDeploymentStatusEvent", DeploymentStatus: app.AppDeploymentStatusEvent{Status: "PENDING"}}) // nolint:errcheck
+			input.Handler(app.DeployEvent{Typename: "AppDeploymentStatusEvent", DeploymentStatus: app.AppDeploymentStatusEvent{Status: "RUNNING"}}) // nolint:errcheck
+		})
+		stdout := os.Stdout
+		r, w, err := os.Pipe()
+		require.NoError(t, err)
+		os.Stdout = w
+		defer func() {
+			os.Stdout = stdout
+		}()
+
+		input := DeployInput{AppDir: appDir, OrgSlug: slug, AppSlug: appSlug, Version: expectedVersion, Message: expectedMessage, Verbose: true}
+		err = Deploy(context.TODO(), apps, input)
+
+		require.NoError(t, w.Close())
+		assert.NoError(t, err)
+		expected := []string{
+			"<nonascii> Loading app configuration............................",
+			"\r<nonascii> Loading app configuration............................OK\n",
+			"<nonascii> Registering new version for organization-slug/app-...",
+			"\r<nonascii> Registering new version for organization-slug/app-...OK\n",
+			"<nonascii> Creating app archive.................................",
+			"\r<nonascii> Creating app archive.................................OK\n",
+			"<nonascii> Uploading app archive................................",
+			"\r<nonascii> Uploading app archive................................OK\n",
+			"<nonascii> Deploying app........................................\n",
+			"Build Build message 1\n",
+			"Build Build message 2\n",
+			"\rDeploy Workload is pending",
+			"\rDeploy Workload is pending.",
+			"\rDeploy Workload is pending..",
+			"\rDeploy Workload is pending..\n",
+			"\rDeploy Workload is running\n",
+			"<nonascii> Deploying app........................................OK\n",
+			"<nonascii> Access your app at: https://www.numerous.com/app/organization/organization-slug/private/app-slug\n",
+		}
+		output, _ := io.ReadAll(r)
+		actual := stripNonASCII(string(output))
+		assert.Equal(t, strings.Join(expected, ""), actual)
+	})
+}
+
+func stripNonASCII(s string) string {
```

Comment:
It might worth to rename the function because it replaces all the characters but `strip` is about leading and trailing characters.

Reply:
I do not have that association with the word "strip". Maybe if I called it "trim", but I'll rename it to `removeNonASCII` instead.

Reply:
It about Python `str.strip`.

Reply:
I changed the name in 1d22137ce6d2b61c11d3471d2caa7aa724490d10